akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -319,15 +320,11 @@ import scala.concurrent.{ Future, Promise }
 
           override def onDownstreamFinish(): Unit = {
             if (!materialized.isCompleted) {
```

Comment:
This conditional is not needed anymore as the only nested action is now `materialized.tryFailure`

Reply:
It's to avoid creating the exception if not needed.

Reply:
(stack traces and all that jazz)

Reply:
+1 for keeping the if